pilot/pkg/model/context.go

```diff
@@ -339,6 +339,25 @@ type WatchedResource struct {
 	// Note that Envoy may send multiple requests for the same type, for
 	// example to update the set of watched resources or to ACK/NACK.
 	LastRequest *discovery.DiscoveryRequest
+
+	// filter is the internal map allowing filters based on ResourceNames
+	filter map[string]struct{}
+}
+
+// IncludesResource allows filtering based on the ResourceNames of a WatchedResource.
+func (w *WatchedResource) IncludesResource(name string) bool {
+	if len(w.ResourceNames) == 0 {
+		return true
+	}
+	if w.filter == nil {
+		// generate the filter only once, assume ResourceNames is always set first and not-changed
```

Comment:
This seems like an assumption that will not be true in many cases. Is this a grpc-only thing that it doesn't change?

Reply:
Yes, but even that could change. I'll just add a util to generate a `Set` and query `empty or contains`